charts/cluster-api-runtime-extensions-nutanix/addons/cluster-autoscaler/values-template.yaml

```diff
@@ -33,3 +33,35 @@ extraVolumeSecrets:
 rbac:
   # Create a Role instead of a ClusterRoles to update cluster-api objects
   clusterScoped: false
+{{- $capiProvider := index .Cluster.Labels "cluster.x-k8s.io/provider" }}
```

Comment:
We talked about this label `cluster.x-k8s.io/provider` in the context of different work and how its technically optional.
Since the example Cluster templates have this label, Im ok with continuing to use it here and in a [couple of other places](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/main/pkg/webhook/cluster/nutanix_validator.go#L61), but we should consider enforcing it exists, or default based on some other information in the Cluster and its templates.

Reply:
Agreed. I've added some docs on this now to hopefully make this easier to use and understand.